### PR TITLE
Tracing with offset

### PR DESCRIPTION
**Pull request by wonder-sk**
This PR improves tracing capabilities when digitizng vector features: it is now possible to set offset when tracing and the traced paths will have the offset applied. Offset can be either positive ("right side) or negative ("left" side).

image

### CODE REVIEW

**Review by nyalldawson — APPROVED** [review]

**Comment by m-kuhn** [comment]
Nice feature :tada:

File: src/core/qgstracer.h [hunk 1]
```diff
@@ -58,6 +58,22 @@ class CORE_EXPORT QgsTracer : public QObject
     //! Set extent to which graph's features will be limited (empty extent means no limit)
     void setExtent( const QgsRectangle &extent );
 
+    //! Get offset in map units that should be applied to the traced paths returned from findShortestPath().
+    //! Positive offset for right side, negative offset for left side.
+    //! \since QGIS 3.0
```
Comment (m-kuhn) [inline src/core/qgstracer.h]: Looks like these docstrings are not picked up by the sip files (onto which I hope we'll soon have a shiny new python api documentation built ;) )

Just guessing that the following format works better

```
/**
 * Docstring
 */
```

**Comment by j-lippmann** [comment]
Could anybody merge  this  PR please ?  I would like test  this nice  feature . Thanks!

**Comment by nyalldawson** [comment]
@wonder-sk has commit rights. I'm sure he'll merge when he knows this PR is ready for master.

File: src/core/qgstracer.h (hunk 1 above)
Comment (m-kuhn) [reply src/core/qgstracer.h]: Confirmed, switching over to multiline comments fixes sipdoc

https://github.com/qgis/QGIS/commit/ab9e2fcb6860912565889f80f324f219a7720de3

**Pull request merged**